src/components/Table/TableCell.tsx

```diff
@@ -17,50 +11,69 @@ export enum TableCellType {
 
 export type TableCellProps = {
     cell: any;
-    state: TableState<any>;
+    selectionMode: string;
     type?: TableCellType;
+    isChecked?: boolean;
+    selectedRows: Key[];
+    setSelectedRows?: (ids?: Key[]) => void;
 };
 
-export const TableCell: FC<TableCellProps> = ({ cell, state, type = TableCellType.Default }) => {
+export const TableCell: FC<TableCellProps> = ({
+    cell,
+    selectionMode,
+    type = TableCellType.Default,
+    isChecked = false,
+    selectedRows,
+    setSelectedRows,
+}) => {
     const ref = useRef<HTMLTableCellElement | null>(null);
-    const { gridCellProps } = useTableCell({ node: cell }, state, ref);
-    const { checkboxProps } = useTableSelectionCheckbox({ key: cell.parentKey }, state);
-    const inputRef = useRef(null);
-    const {
-        inputProps: { checked },
-    } = useCheckbox(checkboxProps, useToggleState(checkboxProps), inputRef);
-    const { isFocusVisible, focusProps } = useFocusRing();
 
     if (type === TableCellType.Checkbox) {
         const { key } = cell;
+        const handleChange = () => {
+            if (!setSelectedRows) {
+                return;
+            }
+
+            if (isChecked) {
+                const filteredRows = selectedRows.filter((row) => row !== cell.parentKey);
+                setSelectedRows(filteredRows);
+                return;
+            }
+            if (selectionMode === SelectionMode.SingleSelect) {
```

Comment:
That last condition could be simplified, WDYT?
```
const selectRows = selectionMode === SelectionMode.SingleSelect ? [cell.parentKey] : [...selectedRows, cell.parentKey];
setSelectedRows(selectRows);
```
giving the variable a better name of course. 😄

Reply:
Sure 👍